File: Programming for Data Science/Final Project/Q2/Q2D/q2d.R
# S3 class `student`, stores the name and age of a student
student <- function(name="", age=0){
    structure(list(name=name, age=age), class="student")
}

# declare getName function
getName <- function(x) UseMethod("getName")
# implement getName function; returns name
getName.student <- function(x){
    return(x$name)
}


# R6 class `Student`, stores the name and age of a student
library(R6)

Student <- R6Class("Student",
    public = list(
        initialize = function(name="", age=0){
            private$name <- name
            private$age <- age
        },
        getName = function() {
            return(private$name)
        }
    ),
    private = list(
        name = NULL,
        age = NULL
    )
)

File: Programming for Data Science/Final Project/Q2/Q2D/q2d.py
class Student:
    """Stores the name and age of a student

    Contains attributes:
      - name: name of the student
      - age: age of the student

    functions:
      - __init__
      - getName()
    """

    def __init__(self, name="", age=0):
        """Initialise moment object"""
        self.name = name
        self.age = age

    def getName(self):
        """Get the name of the student"""
        return self.name
